nmpolicy/internal/resolver/resolver.go

```diff
@@ -0,0 +1,54 @@
+package resolver
+
+import (
+	"fmt"
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/ast"
+)
+
+type Resolver struct {
+	CurrentState map[string]interface{}
+}
+
+func (c *Resolver) Resolve(rootNode ast.Node) (map[string]interface{}, error) {
```

Comment:
The input of resolver is the currentState and capture AST, it has to be like the following

```golang
type Resolver struct {
    currentState map[string]interface{}
    captureAST map[string]ast.Node
}

func (c *Resolver) Resolve() (map[string]interface{}, error) {
```